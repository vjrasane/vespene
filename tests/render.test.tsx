import V, { FunctionComponent, render } from "../src";

describe("render", () => {
  it("renders hello world", () => {
    render(document.body, <div>HELLO WORLD!</div>);
    expect(document.body).toMatchSnapshot();
  });

  it("does not render falsy value", () => {
    const Component = () => null;
    render(document.body, <Component />);
    expect(document.body.childElementCount).toBe(0);
  });

  it("renders component with attributes", () => {
    render(
      document.body,
      <div
        id="id"
        className="class"
        style={{
          cssProperty: "value",
          otherProperty: 1234,
        }}
      >
        Attributes
      </div>
    );
    expect(document.body).toMatchSnapshot();
  });

  it("renders function component", () => {
    const Component = () => <div>Function</div>;

    render(document.body, <Component />);
    expect(document.body).toMatchSnapshot();
  });

  it("renders fragment", () => {
    render(document.body, 
    <>
      <div>Div</div>
      Text
      <span>Span</span>
    </>);
    expect(document.body).toMatchSnapshot();
  });

  it("renders array", () => {
    render(document.body, 
    <div>
      {[
        <div>Div</div>,
        "Text",
        <span>Span</span>,
        [
          <div>Div</div>,
          "Text",
          <span>Span</span>,
          [
            <div>Div</div>,
            "Text",
            <span>Span</span>,
          ]
        ]
      ]}
    </div>);
    expect(document.body).toMatchSnapshot();
  });

  it("does not call render multiple times for mounted component", () => {
    const func = jest.fn();
    const Component = () => { 
      func();
      return <div>Function</div>;
    };

    const elem = <Component />;

    expect(func).not.toHaveBeenCalled();
    render(document.body, elem);
    render(document.body, elem);
    console.log(document.body.innerHTML);
    render(document.body, elem);
    expect(document.body).toMatchSnapshot();
    expect(func).toHaveBeenCalledTimes(1);
  });

  it("renders function component with children", () => {
    const Component: FunctionComponent = ({ children }) => (
      <div>
        <span>Function</span>
        {children}
      </div>
    );

    render(
      document.body,
      <Component>
        <span>Children</span>
      </Component>
    );
    expect(document.body).toMatchSnapshot();
  });
});
