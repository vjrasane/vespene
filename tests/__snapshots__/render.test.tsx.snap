// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render does not call render multiple times for mounted component 1`] = `
<body>
  <div>
    Function
  </div>
</body>
`;

exports[`render renders array 1`] = `
<body>
  <div>
    <div>
      Div
    </div>
    Text
    <span>
      Span
    </span>
    <div>
      Div
    </div>
    Text
    <span>
      Span
    </span>
    <div>
      Div
    </div>
    Text
    <span>
      Span
    </span>
  </div>
</body>
`;

exports[`render renders component with attributes 1`] = `
<body>
  <div
    classname="class"
    id="id"
    style="css-property: value; other-property: 1234"
  >
    Attributes
  </div>
</body>
`;

exports[`render renders fragment 1`] = `
<body>
  <div>
    Div
  </div>
  Text
  <span>
    Span
  </span>
</body>
`;

exports[`render renders function component 1`] = `
<body>
  <div>
    Function
  </div>
</body>
`;

exports[`render renders function component with children 1`] = `
<body>
  <div>
    <span>
      Function
    </span>
    <span>
      Children
    </span>
  </div>
</body>
`;

exports[`render renders hello world 1`] = `
<body>
  <div>
    HELLO WORLD!
  </div>
</body>
`;
